### PR TITLE
Feature/python additions

### PR DESCRIPTION
**Pull request by petersteneteg**
Fixes an issue with the OpenGL format not being set when staring the inviwopyapp.
Added implicit conversions from python list to glm vec types.
Added a implementation of a ipython event loop hook for the inviwopyapp

### CODE REVIEW

File: apps/inviwopyapp/inviwo.py [hunk 1]
```diff
@@ -30,8 +52,13 @@
     app.update()
     app.waitForPool()
     app.update()
+
     # Save a snapshot
-    app.network.Canvas.snapshot("snapshot.png") 
+    #app.network.Canvas.snapshot("snapshot.png")
```
Comment (ResearchDaniel) [inline apps/inviwopyapp/inviwo.py]: Decomment

File: apps/inviwopyapp/inviwo.py [hunk 2]
```diff
@@ -30,8 +52,13 @@
...
     # Save a snapshot
-    app.network.Canvas.snapshot("snapshot.png") 
+    #app.network.Canvas.snapshot("snapshot.png") 
 
     # run the app event loop
-    app.run()
\ No newline at end of file
+    app.run()
+
+    # addIpythonGuiHook(app)
```
Comment (ResearchDaniel) [inline apps/inviwopyapp/inviwo.py]: Decomment or remove

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]
Some minor de/comment issues that need to be fixed

File: apps/inviwopyapp/inviwo.py [hunk 3]
```diff
@@ -30,8 +52,13 @@
...
+
+
```
Comment (martinfalk) [inline apps/inviwopyapp/inviwo.py]: fix newlines at end of file

**Review by martinfalk — DISMISSED** [review]

File: apps/inviwopyapp/inviwo.py (hunk 2 above)
Comment (petersteneteg) [reply apps/inviwopyapp/inviwo.py]: working on adding some cli arguments to the script...

File: include/inviwo/core/util/unindent.h [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+ *     )";
+ * ```
+ * becomes
+ * ```
+ * R"(0123456789
+ * abcdefghij
+ * asdfasdfas
+ * )"
+ * ```
+ * The first indented line determines the amount of indentation to remove, If any of the following
```
Comment (ResearchDaniel) [inline include/inviwo/core/util/unindent.h]: Typos. 
remove, -> remove.
as less indentation -> are less than indentation

File: include/inviwo/core/util/unindent.h [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    StaticString<M> res{};
+    indent::copy(str.begin(), str.end(), res.begin());
+    return res;
+}
+
+}  // namespace indent
+
+/**
+ * Remove the leading indentation from each line of the string.
+ * The first indented line determines the amount of indentation to remove, If any of the following
```
Comment (ResearchDaniel) [inline include/inviwo/core/util/unindent.h]: Same typos as above

**Review by ResearchDaniel — DISMISSED** [review]

**Review by ResearchDaniel — APPROVED** [review]

File: apps/inviwopyapp/inviwo.py [hunk 6]
```diff
@@ -1,64 +1,164 @@
...
+        property list widget
+    :param screenshot: Path to a output file for a screenshot of the 'Canvas' processor
+    :paran run: Start the qt event loop, default False
+    :param ipython: Register the ipython event loop hook. Use the ipython magic command
+        '#gui inviwo' to start.
+    :param list_processors: Output a list of all processors, default False.
+    :param list_properties: Output a list of all properties of the Processors identifiers given
+
+
+    If python can't find the i`inviwopy` module use the '--libdir' CLI argument to pass
```
Comment (martinfalk) [inline apps/inviwopyapp/inviwo.py:42]: special syntax for `i'inviwopy'`?

File: include/inviwo/core/util/unindent.h [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+constexpr size_t length(const StaticString<N>& str) {
+    return length(str.begin(), str.end());
+}
+template <size_t N>
+constexpr size_t length(const char (&str)[N]) {
+    return length(str, str + N - 1);
+}
+
+/**
+ * Copy the chard of the string (being, end) to `out` skipping the leading indentation of each line
```
Comment (martinfalk) [inline include/inviwo/core/util/unindent.h:120]: typo `chars`

**Review by martinfalk — APPROVED** [review]

**Pull request merged**